module/VuFind/src/VuFind/Session/SecureDelegator.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * Secure session delegator
+ *
+ * Copyright (C) Villanova University 2018,
+ *               Leipzig University Library <info@ub.uni-leipzig.de> 2018.
+ *
+ * PHP version 7
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Session_Handlers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Sebastian Kehr <kehr@ub.uni-leipzig.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:session_handlers Wiki
+ */
+namespace VuFind\Session;
+
+use VuFind\Db\Table\PluginManager;
+use Zend\Config\Config;
+use Zend\Crypt\BlockCipher;
+use Zend\Math\Rand;
+
+/**
+ * Secure session delegator
+ *
+ * @category VuFind
+ * @package  Session_Handlers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Sebastian Kehr <kehr@ub.uni-leipzig.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:session_handlers Wiki
+ */
+class SecureDelegator implements HandlerInterface
+{
+    /**
+     * The block cipher for en/decrypting session data.
+     *
+     * @var BlockCipher
+     */
+    protected $cipher;
+
+    /**
+     * The wrapped session handler.
+     *
+     * @var HandlerInterface
+     */
+    protected $handler;
+
+    /**
+     * SecureDelegator constructor.
+     *
+     * @param HandlerInterface $handler {@see $handler}
+     */
+    public function __construct(HandlerInterface $handler)
+    {
+        $this->handler = $handler;
+        $this->cipher = BlockCipher::factory('openssl');
+    }
+
+    /**
+     * Enable session writing (default)
+     *
+     * @return void
+     */
+    public function enableWrites()
+    {
+        $this->handler->enableWrites();
```

Comment:
All the repeated proxying of `$this->handler` makes me wish we could use `__call` to simplify the class... but I guess with the constraints of the interface, that is not technically possible. Too bad!